src/System.Linq.Parallel/tests/ExchangeTests.cs

```diff
@@ -10,183 +11,177 @@ namespace Test
 {
     public class ExchangeTests
     {
-        [Fact]
-        public static void SimplePartitionMergeWhereScanTest()
+        private static readonly ParallelMergeOptions[] Options = new[] {
+            ParallelMergeOptions.AutoBuffered,
+            ParallelMergeOptions.Default,
+            ParallelMergeOptions.FullyBuffered,
+            ParallelMergeOptions.NotBuffered
+        };
+
+        public static IEnumerable<object[]> PartitioningData(object[] counts)
         {
-            for (int i = 1; i <= 128; i *= 2)
-            {
-                SimplePartitionMergeWhereScanTest1(1024 * 8, i, true);
-                SimplePartitionMergeWhereScanTest1(1024 * 8, i, false);
-            }
+            foreach (object[] results in Sources.Ranges(counts.Cast<int>(), x => new[] { 1, 2, 4 })) yield return results;
```

Comment:
I don't think we have a guideline set (though maybe we should).  In general for me it's less about length and more about parsability... there's just a lot going on here (e.g. a loop, a variable being declared, multiple methods being called, LINQ being used, data being yielded, etc.), and I found it more difficult to comprehend what was going on with it on a single line.